sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalogSuite.scala

```diff
@@ -40,6 +40,15 @@ abstract class ExternalCatalogSuite extends SparkFunSuite with BeforeAndAfterEac
 
   protected def resetState(): Unit = { }
 
+  // Clear all state before each test
+  override def beforeEach(): Unit = {
+    try {
+      resetState()
```

Comment:
Since the test cases in `ExternalCatalog` do not drop the created tables, we need to drop it. To avoid duplicate cleanup, we can change `afterEach` to `afterAll`. Let me fix it.